import_models_to_verta/mlflow_databricks_bulk_import_example.ipynb

```diff
@@ -0,0 +1,341 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Import MLflow Models into Verta\n",
+    "\n",
+    "Whether moving models selectively or in bulk, it is possible to import your MLflow models directly into Verta. The following notebook demonstrates how."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Environment Setup"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from typing import Dict, List\n",
+    "try:\n",
+    "    import verta\n",
+    "    from verta.registry.stage_change import Archived, Staging, Production\n",
+    "except ImportError:\n",
+    "    !pip install verta\n",
+    "    import verta\n",
+    "    from verta.registry.stage_change import Archived, Staging, Production\n",
+    "try:\n",
+    "    import mlflow\n",
+    "    from mlflow import MlflowClient\n",
+    "    from mlflow.entities.model_registry import RegisteredModel as MLflow_RegisteredModel\n",
+    "    from mlflow.store.entities.paged_list import PagedList\n",
+    "except ImportError:\n",
+    "    !pip install mlflow\n",
+    "    import mlflow\n",
+    "    from mlflow import MlflowClient\n",
+    "    from mlflow.entities.model_registry import RegisteredModel as MLflow_RegisteredModel\n",
+    "    from mlflow.store.entities.paged_list import PagedList"
```

Comment:
I'd recommend

```python
!pip install verta mlflow
```

```python
import os
from typing import Dict, List

import verta
from verta.registry.stage_change import Archived, Staging, Production

import mlflow
from mlflow import MlflowClient
from mlflow.entities.model_registry import RegisteredModel as MLflow_RegisteredModel
from mlflow.store.entities.paged_list import PagedList
```

to be cleaner, reduce code duplication, and reduce the number of errors we're handling (they can look a bit scary in an example notebook). `pip install` is a no-op if the library is already present, so it's quite harmless to do without nesting it in an `except`.

I'd been meaning to do this for some of our other example notebooks, too.

Reply:
Updated